### ISSUE TITLE
declare -A doesn't work in the default bash on macOS

### ISSUE THREAD

**Issue by augmentedfourth** [nojhan/liquidprompt#554]
https://github.com/nojhan/liquidprompt/blob/6f5af893a3aba29d54005366ba88c3b10c72bcb6/liquidprompt#L226

macOS High Sierra ships with bash 3.2.57, which doesn't understand the `-A` option to `declare`.

I can fix it by setting my default shell to Homebrew bash (4.4.23), but that's a little frustrating.

**Comment by Rycieos** [comment]
I found this hard to believe, but I dug into the change logs: https://tiswww.case.edu/php/chet/bash/CHANGES

```
This document details the changes between this version, bash-4.0-alpha,
and the previous version, bash-3.2-release.
...
ii. The shell provides associative array variables, with the appropriate
    support to create, delete, assign values to, and expand them. (declare -A)
```

We claim to support bash >= 3.2.0:
https://github.com/nojhan/liquidprompt/blob/eda83efe4e0044f880370ed5e92aa7e3fdbef971/liquidprompt#L59-L61

This is important, as Apple won't ship a newer bash than 3.2 for licensing reasons.

However, this change is not in master (yet); it is in commit 3f5af30c257644191fc5a9df08386ed62e853087 on @nojhan's `develop` branch. This is not the officially supported branch. But it brings up a good point.

@nojhan, this change would force us to drop support for bash 3.2, which would force us to drop support for MacOS. Are you planning on merging these develop changes to master?

**Comment by Rycieos** [comment]
Maybe the `devel` version should follow `master` instead? It depends on how @nojhan is planning on doing branches in the future.

On that note, it's been time for a long time for a new official release.

**Comment by augmentedfourth** [comment]
I forked the homebrew packaging discussion to #555 so this thread can get back on topic.

@nojhan, are you planning to deprecate bash 3.2 (and with, it all of macOS, unless people figure out how to manually upgrade bash) by introducing `declare -A` in 1e8a2a19 (and moving it in 3f5af30c)? I hope not. How can we fix this?

**Comment by augmentedfourth** [comment]
Note also that the inclusion of `LP_ENABLE_ERR` in `develop` (added in both ee44113 and e6a4271, oddly, then merged in 8e9bc92) is basically the same change I proposed in 0f4bd8f under #544 (addressing #543). That PR was rejected due to the opinion that @dolmen didn't want to add any new options to `.liquidpromptrc` (and there was an alternate way to fix that particular issue with a custom theme).

It looks like `develop` adds the following new options from what's currently in `master`:
`LP_LOAD_SCALE_MAX`
`LP_ENABLE_RUBY_VENV`
`LP_RUBY_RVM_PROMPT_OPTIONS`
`LP_ENABLE_ERR`
`LP_MARK_SPACE`
`LP_MARK_VCS_OPEN`
`LP_MARK_VCS_CLOSE`
`LP_MARK_X11`
`LP_COLOR_IN_LOCAL`

Of those, only `LP_ENABLE_RUBY_VENV` and `LP_RUBY_RVM_PROMPT_OPTIONS` are added/documented in `liquidpromptrc-dist`.

I'm not sure what to do about any of that; @Rycieos?

**Comment by Rycieos** [comment]
@nojhan is leading the new development of `develop`. I'm not sure where the project stands in terms of what branches are official. `master` has been the main branch for a long time, and I don't think that has changed, so only `master` can be called "released".

As for options added in `develop` not added in `liquidpromptrc-dist`, here is as good as anywhere else to report them. That will need to be completed before `develop` can be merged.

**Comment by dolmen** [comment]
As the recent commits by @nojhan have been merged without references to pull requests, I'm a bit lost about which PR are still relevant and which have been merged. 😞

**Comment by GaryBoone** [comment]
Sorry, I was unable to follow the thread above wrt the status of the bug/fix. I noticed this problem yesterday and have disabled my broken installation of liquidprompt. Is there a release branch with a fix?

**Comment by Rycieos** [comment]
@GaryBoone only the development branch `develop` has this bug, the main release branch `master` does not have it.

**Comment by GaryBoone** [comment]
Ah, I see. It appears that the Homebrew version uses the `develop` branch.

**Comment by augmentedfourth** [comment]
@GaryBoone I actually changed the Homebrew version to `develop` after @nojhan added his recent commits, because I wanted to test them (It had previously been `master` until https://github.com/Homebrew/homebrew-core/pull/31302). Also, the point of a `head` moving target in Homebrew is to follow the most recent bleeding-edge development.

I tried to get Homebrew to take a `devel` version (intended for in-development versions that aren't yet fully released) that follows the liquidprompt `master` branch, but they they rejected it because they don't want `devel` to follow a moving target but instead point at an upcoming version that will be "released soon" (https://github.com/Homebrew/homebrew-core/pull/31302).

It seems, in order to get the current `master` status back into Homebrew, liquidprompt should either prepare a new official release (discussed a bit in #555) or perhaps attempt to re-do the Homebrew pull request to point the `devel` installation at the specific commit hash currently at the tip of `master` rather than a moving target.
